Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -61,14 +66,113 @@ - (BOOL)installWithOptions:(nonnull SentryOptions *)options
 
     [SentryGlobalEventProcessor.shared
         addEventProcessor:^SentryEvent *_Nullable(SentryEvent *_Nonnull event) {
-            [self.sessionReplay captureReplayForEvent:event];
-
+            if (event.isCrashEvent) {
+                [self resumePreviousSessionReplay:event];
+            } else {
+                [self.sessionReplay captureReplayForEvent:event];
+            }
             return event;
         }];
 
     return YES;
 }
 
+- (void)resumePreviousSessionReplay:(SentryEvent *)event
+{
+    NSURL *dir = [self replayDirectory];
+    NSData *lastReplay =
+        [NSData dataWithContentsOfURL:[dir URLByAppendingPathComponent:@"lastreplay"]];
+    if (lastReplay == nil) {
+        return;
+    }
+    NSError *error = nil;
+    NSDictionary<NSString *, id> *jsonObject = [NSJSONSerialization JSONObjectWithData:lastReplay
+                                                                               options:0
+                                                                                 error:&error];
+    if (jsonObject == nil) {
+        SENTRY_LOG_DEBUG(@"Can't open last session replay: %@", error);
+        return;
+    }
+
+    SentryId *replayId = jsonObject[@"replayId"]
+        ? [[SentryId alloc] initWithUUIDString:jsonObject[@"replayId"]]
+        : [[SentryId alloc] init];
+    NSURL *lastReplayURL = [dir URLByAppendingPathComponent:jsonObject[@"path"]];
+
+    SentryCrashReplay crashInfo = { 0 };
+    bool hasCrashInfo = sentrySessionReplaySync_readInfo(&crashInfo,
+        [[lastReplayURL URLByAppendingPathComponent:@"crashInfo"].path
+            cStringUsingEncoding:NSUTF8StringEncoding]);
+
+    SentryReplayType type = hasCrashInfo ? SentryReplayTypeSession : SentryReplayTypeBuffer;
```

Comment:
I've seen you're also persisting `errorSampleRate` as part of crashInfo, I guess you also want to roll the dice for the Buffer mode here based on that persisted sample rate, right?

Reply:
True, true, true. I thought about it. Forgot to use it. Thanks!!